### PR TITLE
Cloud: Document VPC connection via PrivateLink and PSC

### LINKED ISSUE #1518: Document privatelink support in Cloud
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

https://docs.risingwave.com/cloud/intro/

Please identify the topic to be updated and the necessary structure adjustment.

### Reference

@wyhyhyhyh Please provide any information that you have. Thanks.

### PR DESCRIPTION
**Pull request by CharlieSYH**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Doc impact**

  - VPC connection 
      - Overview (concept, diagram, subpage selection): https://pr-1536.d2fbku9n2b6wde.amplifyapp.com/cloud/vpc-overview/
      - Create a VPC connection (steps): https://pr-1536.d2fbku9n2b6wde.amplifyapp.com/cloud/create-a-connection/
      - Drop a VPC connection (steps): https://pr-1536.d2fbku9n2b6wde.amplifyapp.com/cloud/drop-a-connection/
  - Some updates to the kernel docs.

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1518

## For reviewers

- **Key points**

  - @cyliu0 Please provide a screenshot and steps for retrieving the server target URL of the service attachment since I don't have a GCP account.

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: cloud/vpc-create-a-connection.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    alt="Create connection page"
+    />
+
+2. For **Connection type**, select your cloud service provider. 
+   
+   Currently, RisingWave Cloud supports **AWS** PrivateLink and **GCP** Private Service Connect.
+
+3. For **Cluster**, select the cluster you want to connect the VPC to.
+   
+   Notice that the VPC and the cluster must be in the same region.
```
Comment (emile-00) [inline cloud/vpc-create-a-connection.md]: "Ensure" instead of "Notice" might work better here.

**Comment by cyliu0** [comment]
By the way, should we mention that we can only support unauthenticated access like MSK for now in this doc?

image

File: cloud/vpc-create-a-connection.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+:::
+
+After you created the connection, a VPC connection endpoint is generated for your cluster. You can find it in [**Connection**](https://cloud.risingwave.com/connection/).
+
+<img
+src={require('./images/vpc-connection-endpoint.png').default}
+alt="VPC connection endpoint "
+/>
+
+For details on how to use the VPC endpoint to create a source with the VPC connection, see [Create source with VPC connection](https://docs.risingwave.com/docs/current/ingest-from-kafka/#create-source-with-vpc-connection). For creating a sink, see [Create sink with VPC connection](https://docs.risingwave.com/docs/current/create-sink-kafka/#create-sink-with-vpc-connection).
```
Comment (cyliu0) [inline cloud/vpc-create-a-connection.md:83]: The links are wrong. Should be:
- https://docs.risingwave.com/docs/current/ingest-from-kafka/#create-source-with-aws-privatelink-connection
- https://docs.risingwave.com/docs/current/create-sink-kafka/#create-sink-with-aws-privatelink-connection

Comment (CharlieSYH) [reply cloud/vpc-create-a-connection.md:83]: Actually, the links will be updated to the ones I used in the draft. But thanks for noticing that!

File: cloud/images/gcp-service-attachment.png
Comment (wyhyhyhyh) [inline cloud/images/gcp-service-attachment.png:1]: Need mosaic on: 
- The title `cy-kafka-service`
- Connected projects

File: cloud/images/vpc-diagram.png
Comment (wyhyhyhyh) [inline cloud/images/vpc-diagram.png:1]: Can we ask Nika to re-depict this? We need some inspiration from designers

File: cloud/vpc-create-a-connection.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
+---
+id: vpc-create-a-connection
+title: Create a VPC connection
+description: Create a VPC connection through PrivateLink or Private Service Connect.
+slug: /create-a-connection
+---
+
+Follow the steps below to establish a secure connection with your VPC through AWS PrivateLink or GCP Private Service Connect.
+
+## Prerequisites
```
Comment (wyhyhyhyh) [inline cloud/vpc-create-a-connection.md:10]: We may not want to say pre-requisites. Instead, we add a "restriction" section and put it there.

Comment (wyhyhyhyh) [reply cloud/vpc-create-a-connection.md:10]: Just a suggestion, we can start with this version first.

File: cloud/vpc-overview.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
+id: vpc-overview
+title: VPC connection
+description: Manage VPC connections.
+slug: /vpc-overview
+---
+
```
Comment (wyhyhyhyh) [inline cloud/vpc-overview.md:7]: Reference: https://docs.confluent.io/cloud/current/networking/private-links/aws-privatelink.html

Comment (wyhyhyhyh) [inline cloud/vpc-overview.md:7]: Consider either adding a section title "concept", or add a separate page titled "concept".

**Review by wyhyhyhyh — APPROVED** [review]

**Pull request merged**